fragmenter/fragment.py

```diff
@@ -19,6 +19,113 @@
 OPENEYE_VERSION = oe.__name__ + '-v' + oe.__version__
 
 
+def expand_states(molecule, protonation=True, tautomers=False, stereoisomers=True, consider_aromaticity=True, maxstates=200,
+                  verbose=True, filename=None):
+    """
+    Expand molecule states (choice of protonation, tautomers and/or stereoisomers).
+    Protonation states expands molecules to protonation of protonation sites (Some states might only be reasonable in
+    very high or low pH. ToDo: Only keep reasonable protonation states)
+    Tatutomers: Should expand to tautomer states but most of hte results are some resonance structures. Defualt if False
+    for this reason
+    Stereoisomers expands enantiomers and geometric isomers (cis/trans).
+    Returns set of SMILES
+
+    Parameters
+    ----------
+    molecule: OEMol
+        Molecule to expand
+    protonation: Bool
+        If True will enumerate protonation states. Default True
+    tautomers: Bool
+        If True, will enumerate tautomers. Default False (Note: This is False because results usually give resonance structures
+        which ins't needed for torsion scans
+    stereoisomers: Bool
+        If True will enumerate stereoisomers (cis/trans and R/S). Default is True
+    consider_atomaticity: Bool
+    maxstates: int
+        Defulat 200
+    verbose: Bool
+    filename: str
+        Filename to save SMILES to. If None, SMILES will not be saved to file. Defualt None.
+
+    Returns
+    -------
+    states: set of SMILES for enumerated states
+
+    """
+    title = molecule.GetTitle()
+    states = set()
+    molecules = [molecule]
+    if verbose:
+        logger().info("Enumerating states for {}".format(title))
+    if protonation:
+        molecules.extend(_expand_states(molecules, enumerate='protonation', consider_aromaticity=consider_aromaticity,
+                                   maxstates=maxstates, verbose=verbose))
+    if tautomers:
+        molecules.extend(_expand_states(molecules, enumerate='tautomers', consider_aromaticity=consider_aromaticity,
+                                   maxstates=maxstates, verbose=verbose))
+    if stereoisomers:
+        molecules.extend(_expand_states(molecules, enumerate='stereoisomers', maxstates=maxstates, verbose=verbose))
+
+    if filename:
+        oname = filename
+        ofs = oechem.oemolostream()
+        if not ofs.open(oname):
```

Comment:
Do you actually use `ofs` to write molecules? I can't see where that is happening here. I just see you open it and then let the handle expire by gc.

Reply:
Thanks for finding this. I had used this in the a previous version of the code.